src/platform/windows/windows_ime.c

```diff
@@ -60,14 +60,39 @@ static void WinIME_OnChar(LCUI_Event e, void *arg)
 	LCUIIME_Commit(text, 2);
 }
 
+static void WinIME_OnCaretPosChanged(LCUI_SysEvent e, void *arg)
+{
+	HWND hwnd = GetActiveWindow();
+	HIMC hIMC = ImmGetContext(hwnd);
+	LCUI_Pos caret_pos;
+	if (hIMC) {
+		POINT point;
+		caret_pos = LCUI_GetCaretPos();
+		point.x = caret_pos.x;
+		point.y = caret_pos.y;
+
+		COMPOSITIONFORM Composition;
+		Composition.dwStyle = CFS_POINT;
+		Composition.ptCurrentPos.x = point.x;
+		Composition.ptCurrentPos.y = point.y;
+		ImmSetCompositionWindow(hIMC, &Composition);
+
+		ImmReleaseContext(hwnd, hIMC);
+	}
+}
+
+static int OnCaretPosChanged_Id;
```

Comment:
All global variables should be centrally defined at the top, you should move it to the front of IME_ProcessKey()